### ISSUE TITLE
Keyboard shortcut for switching preview/edit

### ISSUE THREAD

**Issue by psy-q** [nextcloud/notes#505]
**Is your feature request related to a problem? Please describe.**
Switching between editor and preview requires switching hands to the mouse and then going into a drop-down menu where the middle item needs to be selected. This is quite cumbersome when you're in deep in a writing flow and just want a quick preview.

**Describe the solution you'd like**
Define a keyboard shortcut (maybe Alt+P) that switches between edit and preview modes.

**Describe alternatives you've considered**
Display the preview on the side of the Markdown editor.

**Comment by korelstar** [comment]
Sound like a good enhancement. @jancborchardt which keyboard shortcut is recommendable for this? How could we advertise this?

**Comment by korelstar** [comment]
Btw.: this issue is kind-of related to #366.

**Comment by jancborchardt** [comment]
> Sound like a good enhancement. @jancborchardt which keyboard shortcut is recommendable for this? How could we advertise this?

Good question, there’s not really a standard:
- [Github comments use `Ctrl Shift P`](https://help.github.com/en/github/getting-started-with-github/keyboard-shortcuts#comments) but this conflicts with "New Private Window"
- Same for [Simplenote using `Ctrl Shift P`](https://github.com/Automattic/simplenote-electron/pull/1983)
- [Atom’s Markdown preview uses `Ctrl Shift M`](https://github.com/atom/markdown-preview#markdown-preview-package) which in GNOME at least triggers the magnifying glass accessibility feature.
- [iA Writer for macOS uses `Command R`](https://ia.net/writer/support/general/markdown-guide) (or swiping) which of course on the web would conflict with refreshing the page
- [Visual Studio Code’s Markdown preview uses `Ctrl Shift V`](https://code.visualstudio.com/docs/languages/markdown#_markdown-preview) which seems to not conflict with anything

**Short version: Out of all of them, [Visual Studio Code’s Markdown preview uses `Ctrl Shift V`](https://code.visualstudio.com/docs/languages/markdown#_markdown-preview) is the only one which does not conflict with anything existing and is also a widely used software, so let’s go for that one.**

**Comment by stefan-niedermann** [comment]
ctrl shift v is often used to paste formatted text *without* the formatting. Not sure if this is relevant here, just wanted to note.

**Comment by korelstar** [comment]
Thanks for the overview, @jancborchardt . But due to the comment by @stefan-niedermann , I'm confused, now 😕 . Should be still use `Ctrl Shift V`? How could we advertise this to the user?

**Comment by jancborchardt** [comment]
I’d say let’s go for `Ctrl Shift V` as that is used by Visual Studio Code – it seems quite difficult here to find any shortcut which at the same time is established and does not conflict with anything, so let’s just go with that. Possibly in the future a de-facto standard will evolve. :)

How we usually advertise shortcuts is in the text or tooltips of the buttons. Since it is a longer shortcut, it definitely needs to be in the tooltip.

**Comment by ecogit** [comment]
> > Sound like a good enhancement. @jancborchardt which keyboard shortcut is recommendable for this? How could we advertise this?
> 
> Good question, there’s not really a standard:
> 
>     * [Github comments use `Ctrl Shift P`](https://help.github.com/en/github/getting-started-with-github/keyboard-shortcuts#comments) but this conflicts with "New Private Window"
> 
>     * Same for [Simplenote using `Ctrl Shift P`](https://github.com/Automattic/simplenote-electron/pull/1983)
> 
>     * [Atom’s Markdown preview uses `Ctrl Shift M`](https://github.com/atom/markdown-preview#markdown-preview-package) which in GNOME at least triggers the magnifying glass accessibility feature.
> 
>     * [iA Writer for macOS uses `Command R`](https://ia.net/writer/support/general/markdown-guide) (or swiping) which of course on the web would conflict with refreshing the page
> 
>     * [Visual Studio Code’s Markdown preview uses `Ctrl Shift V`](https://code.visualstudio.com/docs/languages/markdown#_markdown-preview) which seems to not conflict with anything
> 
> 
> **Short version: Out of all of them, [Visual Studio Code’s Markdown preview uses `Ctrl Shift V`](https://code.visualstudio.com/docs/languages/markdown#_markdown-preview) is the only one which does not conflict with anything existing and is also a widely used software, so let’s go for that one.**

[Typora](https://support.typora.io/Shortcut-Keys/) is a popular markdown editor among macOS users (yet also available for Windows and Linux) because of its slick design and powerful functionality (table import, mermaid, typewriter mode, etc.). They use ctrl+/ (forward slash; to go from default preview-like mode to 'source code mode' in Markdown syntax. So far I haven't faced complications with global shortcuts (Ubuntu 18.04).

**Comment by stefan-niedermann** [comment]
Agree, `ctrl` + `/` sounds good, indeed. I only know that it's used by atlassian apps like Jira and Confluence for search, but that shouldn't affect us at all.
